### PR TITLE
Employee dashboard

### PR DESCRIPTION
**Pull request by C-Ivan-Bandilla**
Found the issue in assets but not resolve 

after reading some forums livewire navigate updates daw the body only

but after digging through the webpage found out that there are duplicate conflicting style injected by both the livewire and vite

https://drive.google.com/file/d/1o-zEfEq3WN_RY2bAIOsDTQQtm_3mzaUo/view?usp=sharing

### CODE REVIEW

**Comment by C-Ivan-Bandilla** [comment]
Updated factories is in here to [2a3d1b1](https://github.com/POWERLANE-TEAM/emplo-hrms/pull/33/commits/2a3d1b146cc3f933c9cd657e72e8b87bd20542ce)

**Comment by C-Ivan-Bandilla** [comment]
Fixed the bug on wire:navigate, instead of importing CSS inside js file I added as external style and script separately
image


https://drive.google.com/file/d/1o0dGGuVLzJMI8b0y_5-v6fb7hjJOEa2p/view?usp=sharing

**Comment by Carl-Tabuso** [comment]
about slow serving of assets, i'm not sure if it's only in dev server. The delay is very noticeable when navigating through different components, e.g. index to sign-in or vice-versa

**Comment by Carl-Tabuso** [comment]
Probably because the assets are not yet built for prod.

**Comment by Carl-Tabuso** [comment]
Can you try to rewrite the git history of this pr? Gitguardian will fail in every pr once this is merged.

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                break;
+            case 'SYSADMIN':
+                return view('employee.head-admin.index', ['user' => $user_with_role_and_account]);
+                break;
+            case 'APPLICANT':
+                return redirect()->to('/');
+                break;
+            default:
+                return view('employee.standard.index', ['user' => $user_with_role_and_account]);
+        }
```
Comment (Carl-Tabuso) [inline app/Http/Controllers/DashboardController.php:36]: I get a warning from the compiler on `break` of every switch-case. You can simply remove the `break` as it is "Unreachable", the execution terminates if a return statement is encountered.

image

File: app/Http/Controllers/PreEmploymentController.php
Comment (Carl-Tabuso) [inline app/Http/Controllers/PreEmploymentController.php:1]: a bit of suggestion for consistency: Stick with the dot notation when rendering blade components.

image

File: app/Livewire/Auth/SignUp.php
Comment (Carl-Tabuso) [inline app/Livewire/Auth/SignUp.php:1]: for validation using livewire component, i suggest following the sample code below for easy familiarity and cleanliness.For more info, check [here.](https://livewire.laravel.com/docs/validation)

```php
<?php
 
namespace App\Livewire\Forms;
 
use Livewire\Attributes\Validate;
use Livewire\Form;
 
class PostForm extends Form
{
    #[Validate('required|min:3')]
    public $title = '';
 
    #[Validate('required|min:3')]
    public $content = '';

    // then you may simply say for example here
    public function save()
    {
        $this->validate(); // use this to enforce the validation and you may also replace this with the validateOnly()  method to only enforce validation rules on certain properties, e.g. $this->validateOnly('title')
             
         Post::create([
                'title' => $this->title,
                'content' => $this->content,
         ]);
        return redirect()->to('/posts');
    }
}
```

Comment (C-Ivan-Bandilla) [reply app/Livewire/Auth/SignUp.php:1]: Okay will try to address on a different pr

File: app/Http/Controllers/DashboardController.php (hunk 1 above)
Comment (C-Ivan-Bandilla) [reply app/Http/Controllers/DashboardController.php]: addressed in 9042549

File: app/Http/Controllers/PreEmploymentController.php
Comment (C-Ivan-Bandilla) [reply app/Http/Controllers/PreEmploymentController.php:1]: addressed in a11dbbe

**Comment by C-Ivan-Bandilla** [comment]
> Gitguardian detects the logs of @Cedieziah debugbar. Can you try to rewrite the git history of this pr? Gitguardian will fail in every pr once this is merged.

Tried to rebase and removed the logs, the gitguardian now detects no secret

**Pull request closed without merging**